File: frontend/cli/src/app/personas/persona.ts
import { Designacion } from "../designacion/designacion";

export interface Persona {
    dni: number;
    cuil: string;
    nombre: string;
    apellido: string;
    titulo: string;
    sexo: string;
    domicilio: string;
    telefono: string;
    designaciones?: Designacion[];
}

File: backend/src/main/java/unpsjb/labprog/backend/model/Persona.java
package unpsjb.labprog.backend.model;

import java.util.Collection;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Persona {
    @Id
    private int dni;

    private String cuil;
    private String nombre;
    private String apellido;
    private String titulo;
    private char sexo;
    private String domicilio;
    private String telefono;

    @OneToMany(mappedBy = "persona")
    private Collection<Designacion> desiganciones;

}
